### ISSUE TITLE
Should `layout` and `animate` methods of Graph trait consume `self`?

### ISSUE THREAD

**Issue by martinitus** [martinitus/rs-plode#1]
https://github.com/martinitus/rs-plode/blob/a114c960a492d2a04dd8029864c813ff0fa2c0a8/src/lib.rs#L41

Currently, the two mentioned methods take the graph by reference, which binds the lifetime of the returned layout structure to the lifetime of the graph.

This causes problems, if a user is mostly interested in the layout and does not want to manage the lifetime of the used graph explicitly. E.g. :

```rust
fn get_some_layout()->Layout<'a, SomeGraph>{
   let graph = SomeGraph::default();
   // This won't work, as layout is bound to lifetime of graph
   return graph.layout(SomeEngine{}); 
}
```

On the other hand, if the `layout` call consumes the graph, it will not be possible to create multiple layouts without cloning the graphs (also bad).

A solution should allow both, multiple layouts for the same graph and a way to avoid the lifetime bounds.

**Comment by martinitus** [comment]
Turns out, that #2 needs to be fixed before this..

**Comment by martinitus** [comment]
Resolved with #3
